### PR TITLE
Add dumux.sh script to test dumux setup

### PR DESCRIPTION
**Pull request by timokoch**
Provides the Python package dumux with is outside the dune namespace

### CODE REVIEW

**Comment by adedner** [comment]
I can not figure out how to run the action on a fork - your fork doesn't have the actions and the drop down menu to choose the branch when triggering the workflow does not show your fork.... So I'll need to merge before testing :-(

Before I do that: 
the only thing that is missing is a cd into `dumus`.

I don't think shared is default at the moment. Adding something like
`export DUNE_CMAKE_FLAGS=-DBUILD_SHARED_LIBS=ON`
at the beginning of the script should probably do the trick. Once you've make those two changes I'll merge into maom and then test - or try to figure out a better way of doing testing on forks...

File: testing/dumux.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+# clone dumux
+echo "cloning dumux master"
+clonemodule dumux "$dumux_url" master master
+echo "done"
+
+# configure
+dunecontrol --only=dumux all
+
+pushd test/python
```
Comment (timokoch) [inline testing/dumux.sh:24]:
```suggestion
pushd dumux/test/python
```

File: testing/dumux.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# this tests dumux as external package which provides
+# a Python package outside the dune namespace
+
```
Comment (timokoch) [inline testing/dumux.sh:3]: export DUNE_CMAKE_FLAGS=-DBUILD_SHARED_LIBS=ON

File: testing/dumux.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+dumux_url="https://git.iws.uni-stuttgart.de/dumux-repositories/dumux.git"
```
Comment (timokoch) [inline testing/dumux.sh:4]:
```suggestion
# dumux Python bindings currently only work with shared libs
export DUNE_CMAKE_FLAGS=-DBUILD_SHARED_LIBS=ON
dumux_url="https://git.iws.uni-stuttgart.de/dumux-repositories/dumux.git"
```

**Comment by timokoch** [comment]
Maybe better to copy the script then and open an MR directly in the repo? I don't have rights for that hence the fork

**Comment by adedner** [comment]
I'll set up a new branch and copy the file there. I just tested locally and run into the problem
```
Traceback (most recent call last):
  File "test_1p.py", line 87, in <module>
    spatialParams = SpatialParams()
  File "/tmp/DUNE/dumux/build-cmake/python/dumux/porousmediumflow/__init__.py", line 144, in createFVSpatialParamsOneP
    return module.FVSpatialParamsOneP(gridGeometry, cls())
RuntimeError: Dumux::ParameterException [get:/tmp/DUNE/dumux/dumux/common/loggingparametertree.hh:423]: Key Problem.EnableGravity not found in the parameter tree
```
Is that an issue with the shared libs?

You can test the script locally quite easily with this type of script by directly running `bash dumux.sh` in the testing folder, that will use the prerelease pypi versions of the core modules.
Two things I noticed: you need to install `requests` together with `mpi4py` and dumux seems to need `dune-alugrid` so that needs to be added to the `pip install` command with the core modules.
With those two changes dumux was correctly configured and the test started but then the above error occurred. First run of the script takes some time bet then the dune wheels are cached and the next `bash dumux.sh` should be a lot quicker...

**Comment by samuelburbulla** [comment]
@timokoch Thanks for adding this test! This will be very helpful for future changes.

As this repository is getting more important, I propose to move this into the dune-project Github namespace:

https://github.com/dune-project/dune-testpypi

There, I opened a PR with the dumux script and started the `testing scenarios` action.

@adedner The error you observe is likely to be related to the shared libs issue. Haven't we had a similar problem with the parameter class in Fem?

**Comment by adedner** [comment]
@samuelburbulla  that link gives me a 404 page.

I just retested locally again (after removing the cache that there is not surprisingly the old issue with using shared libs aftrer packaging: `ImportError: libdunecommon.so: cannot open shared object file: No such file or directory`
So dumux will not work with  the packages and we will have to switch to using the source modules for core.
This is similar to what is done in `testing/completeSourceInEnv.sh` (only a bit harder to test locally then).

**Comment by samuelburbulla** [comment]
> @samuelburbulla that link gives me a 404 page.

Sorry, I just changed repository's visibility.

**Comment by adedner** [comment]
This way I would lose access to the repo - I don't know if I like that :-)

**Comment by samuelburbulla** [comment]
> This way I would lose access to the repo - I don't know if I like that :-)

I added you to the dune-project owners. Do you have your desired access now?

**Comment by adedner** [comment]
add to dune version of repo

**Pull request closed without merging**